data/config-schema/src/main/resources/net/corda/schema/configuration/crypto/1.0/corda.crypto.json

```diff
@@ -5,437 +5,195 @@
+              ]
             }
-          },
-          "additionalProperties": false
-        },
-        "flow": {
-          "description": "Settings for crypto flow ops messages processor",
+          }
+        }
+      }
+    },
+    "hsm": {
+      "description": "Settings and capabilities for the HSM",
+      "type": "object",
+      "default": {},
+      "properties": {
+        "retry": {
+          "description": "Retry settings for the HSM",
           "type": "object",
           "default": {},
           "properties": {
             "maxAttempts": {
-              "description": "Maximum attempts to process a message",
+              "description": "Maximum number of attempts",
               "type": "integer",
               "default": 3
             },
-            "waitBetweenMills": {
-              "description": "Time between attempts in milliseconds, if the number of values is less than attempts then the last item is repeated",
-              "type": "array",
-              "minItems": 1,
-              "default": [
-                200
-              ],
-              "items": {
-                "type": "integer"
-              }
+            "attemptTimeoutMills": {
+              "description": "Wait period in milliseconds between attempts, should be reasonably large as some operations may be long, e.g. RSA key generation by the SOFT HSM may take a few seconds",
+              "type": "integer",
+              "default": 20000
             }
           },
           "additionalProperties": false
         },
-        "registration": {
-          "description": "Settings for HSM registration messages processor",
+        "categories": {
+          "description": "Categories which the HSM can be used for",
+          "type": "array",
+          "minItems": 1,
+          "items": {
+            "type": "object",
+            "properties": {
+              "category": {
+                "description": "Category name, could be the wildcard * meaning all which has to be the last item if used",
+                "type": "string"
+              },
+              "policy": {
+                "description": "Defines how private keys are generated for that category, ALIASED - stored in the HSM, WRAPPED - wrapped by the HSM, BOTH - decision is taken by the HSM at runtime",
+                "enum": [
+                  "ALIASED",
+                  "WRAPPED",
+                  "BOTH"
+                ]
+              }
+            },
+            "required": [
+              "category",
+              "policy"
+            ],
+            "additionalProperties": false
+          }
+        },
+        "masterKeyPolicy": {
+          "description": "Wrapping key policy, NONE - no wrapping key is required, SHARED - wrapping key shared by all tenants, UNIQUE - wrapping key unique for each tenant",
+          "enum": [
+            "NONE",
+            "SHARED",
+            "UNIQUE"
+          ]
+        },
+        "masterKeyAlias": {
```

Comment:
I think this could be removed and be a hardcoded default for now (happy to be a follow-up PR).

Reply:
Seems to not be used anywhere, will remove.

Reply:
Removed in [#1047](https://github.com/corda/corda-api/pull/1047)